src/ILCompiler.WebAssembly/src/Compiler/WebAssemblyCodegenCompilation.cs

```diff
@@ -17,6 +18,8 @@ public sealed class WebAssemblyCodegenCompilation : Compilation
         internal WebAssemblyCodegenConfigProvider Options { get; }
         internal LLVMModuleRef Module { get; }
         public new WebAssemblyCodegenNodeFactory NodeFactory { get; }
+        internal LLVMDIBuilderRef DIBuilder { get; }
+        internal Dictionary<string, DebugMetadata> DebugMetadataMap { get; }
```

Comment:
Nit: if I was doing this, I would try to wrap the related functionality into a class, instead of hanging a dictionary from a string to a mutable class here. Classes tend to be easier to reason about.

Reply:
I made the class immutable, but I didn't have a good model for wrapping it.